test/jdk/java/util/UUID/UUIDTest.java

```diff
@@ -146,6 +149,83 @@ private static void nameUUIDFromBytesTest() throws Exception {
         }
     }
 
+    private static void epochMillisTest() throws Exception {
+        List<UUID> collisions = new ArrayList<>();
+        Set<UUID> set = new HashSet<>();
+        long now = System.currentTimeMillis();
+
+        for (int i = 0; i < COUNT; i++) {
+            long ts = now + i;
+            UUID u = UUID.epochMillis(ts);
+            if (u.version() != 7) {
+                throw new Exception("Bad version: " + u);
+            }
+            if (u.variant() != 2) {
+                throw new Exception("Bad variant: " + u);
+            }
+            if (!set.add(u)) {
+                collisions.add(u);
+            }
+        }
+
+        if (!collisions.isEmpty()) {
+            // This is extremely unlikely to happen. If you see this failure,
+            // this highly likely points to the implementation bug, rather than
+            // the odd chance.
+            throw new Exception("UUID collisions detected: " + collisions);
+        }
+    }
+
+    private static void epochMillisTest_Multi() throws Exception {
+        long now = System.currentTimeMillis();
+        List<UUID> uuids = IntStream.range(0, COUNT).parallel()
+                .mapToObj(i -> UUID.epochMillis(now + 1))
+                .toList();
+
+        List<UUID> collisions = new ArrayList<>();
+        Set<UUID> set = new HashSet<>();
+        for (UUID u : uuids) {
+            if (u.version() != 7) {
+                throw new Exception("Bad version: " + u);
+            }
+            if (u.variant() != 2) {
+                throw new Exception("Bad variant: " + u);
+            }
+            if (!set.add(u)) {
+                collisions.add(u);
+            }
+        }
+
+        if (!collisions.isEmpty()) {
+            // This is extremely unlikely to happen. If you see this failure,
+            // this highly likely points to the implementation bug, rather than
+            // the odd chance.
+            throw new Exception("UUID collisions detected: " + collisions);
+        }
+    }
```

Comment:
These tests are no longer relevant since the caller is responsible for uniqueness.